nameko/web/handlers.py

```diff
@@ -0,0 +1,108 @@
+import sys
+from logging import getLogger
+from functools import partial
+
+from eventlet.event import Event
+
+from werkzeug.wrappers import Response
+from werkzeug.routing import Rule
+
+from nameko.exceptions import serialize, BadRequest
+from nameko.extensions import Entrypoint
+from nameko.web.server import WebServer
+
+
+_log = getLogger(__name__)
+
+
+def response_from_result(result):
+    if isinstance(result, Response):
+        return result
+
+    headers = None
+    if isinstance(result, tuple):
+        if len(result) == 3:
+            status, headers, payload = result
+        else:
+            status, payload = result
+    else:
+        payload = result
+        status = 200
+
+    return Response(
+        unicode(payload),
+        status=status,
+        headers=headers,
+    )
+
+
+class HttpRequestHandler(Entrypoint):
+    server = WebServer()
+
+    def __init__(self, method, url, expected_exceptions=()):
+        self.method = method
+        self.url = url
+        self.expected_exceptions = expected_exceptions
+
+    def get_url_rule(self):
+        return Rule(self.url, methods=[self.method])
+
+    def setup(self):
+        self.server.register_provider(self)
+
+    def stop(self):
+        self.server.unregister_provider(self)
+        super(HttpRequestHandler, self).stop()
+
+    def process_request_data(self, request):
+        if request.method == 'POST':
+            data = request.get_data()
+            args = (data,)
+        else:
+            args = ()
+
+        kwargs = request.path_values
+        return args, kwargs
+
+    def handle_request(self, request):
+        request.shallow = False
+        try:
+            context_data = self.server.context_data_from_headers(request)
+            args, kwargs = self.process_request_data(request)
+            result = self.handle_message(context_data, args, kwargs)
+            response = response_from_result(result)
+
+        except Exception as exc:
+            exc_info = sys.exc_info()
+            _log.error('request handling failed', exc_info=exc_info)
```

Comment:
Lost my previous comment on this. We don't log this type of error in the other entrypoints. ``error`` seems severe.